docs/DesignDoc.md

```diff
@@ -45,5 +45,15 @@ This section describes the application architecture.
 
 
 ### Class Diagram
-> _Place a class diagram here._
-> _Describe your class diagram._
+> _![class-diagram.png](..%2Fassets%2Fclass-diagram.png)_
+> _This class diagram describes a betting system with several interconnected classes, illustrating the associations between classes and defines the roles each class handles. The diagram illustrates one-to-many relationships between User and Bet, User and Leaderboard, and Event and Bet. The relationship between User and Profile is depicted as one-to-one, indicating that each user has one profile.
+Description of the diagram:
+Profile Class: The Preference field stores user preferences for the application, such as betting options. The userID field contains a direct relationship to the User class, signifying that each profile is associated with a unique user.
+User Class: Describes a user with fields for userID, username, password, and points. It has methods for managing user profiles, checking the leaderboard, and placing bets. This is the central entity that connects to the Profile, Bet, and Leaderboard classes, indicating that users have profiles, can participate in betting, and have standings on the leaderboard.
+Bet Class: Includes fields such as betID, userID, eventID, and result. The result field is a Boolean, which indicates whether the bet was won or lost. This class is associated with the User and Event classes, where a user can have multiple bets, and each bet is linked to a specific event.
```

Comment:
Can we make the different classes formatted with bold text? Example: "**Bet Class:** Includes fields ..."